plugins/module_utils/network/vyos/rm_templates/ntp_global.py

```diff
@@ -35,11 +35,11 @@ def __init__(self, lines=None, module=None):
             "name": "allow_clients",
             "getval": re.compile(
                 r"""
-                ^set\ssystem\sntp\sallow-clients\saddress (\s(?P<ipaddress>\S+))?
+                ^set\s(?P<path>system|service)?\sntp\sallow-clients\saddress (\s(?P<ipaddress>\S+))?
                 $""",
                 re.VERBOSE,
             ),
-            "setval": "system ntp allow-clients address {{allow_clients}}",
+            "setval": "path ntp allow-clients address {{allow_clients}}",
```

Comment:
Do these `path` references need to be `{{path}}`? I wouldn’t expect the variable substitution to work without it.

Reply:
I am still working on more elegant way of abstraction version discrepancies, so I appreciate there might be a better way. Meantime, this is a placeholder, so I can then replace path by either 'system' or 'service' in the resultant self.commands list - keep looking though

Reply:
Yes, I figured, was just wondering if you wanted to use the replacement tool that was already being used to replace the `path`.

Reply:
I may have missed in myriad of lines of code - can you point me to any existing solutions you, guys, already have? thanks

Reply:
If I am to speculate and preempt here, and the replacement tool is something that I coded, then {{path}}?  is getval part and required for changed status checks, otherwise, if only the replacement is left, then it will always be changed = true

Reply:
> Do these `path` references need to be `{{path}}`? I wouldn’t expect the variable substitution to work without it.

No, it did not work for me - that said, I am still checking if the solution can be down purely in the parser code

Reply:
Correct, it’s not a mechanism that will pull directly from the current namespace, you would need to insert the `path` into the dictionary that is being rendered, which might be tricky to find the right place. There may not be a suitable way to inject the data into the underlying render code. 

If not, I’d recommend that you consider something a bit more distinct, maybe “%%path%%” as the token you’re trying to replace  in order to ensure that the code doesn’t replace a portion of an argument (such as a hostname) and create an unexpected result if an NTP host is named “tick.empath.com” for example. That would become “tick.emsystem.com” or “tick.enservice.com” using the current code.